guides/dsl.md

```diff
@@ -0,0 +1,241 @@
+
+```javascript
+program.commit("Assign a letter grade.", ({find, choose}) => {
+    let student = find("student");
+    let [grade] = choose(
+        () => { student.grade >= 90; return "A"; },
+        () => { student.grade >= 80; return "B"; },
+        () => { student.grade >= 70; return "C"; },
+        () => { student.grade >= 60; return "D"; },
+        () => "F"
+    )
+    return [
+        student.add("letter-grade", {grade}),
+    ];
+});
+```
+
+### union()
+
+By contrast `union()` takes a body of branches and combines their output into a single variable. One use of union might be to normalize records from different data sources.
+
+```javascript
+program.block("display the student's full names", ({find, record, union}) => {
+  let east = find("student", {school: "West HS"});
+  let west = find("student", {school: "East HS"});
+  let [fullName] = union(
+    () => { east.name; return east.name},
+    () => { west.firstName; return `${{west.firstName}} ${{west.lastName}}`},
+  );
+  return [
+    record("html/element", {tagname: "div", text: fullName}),
+  ];
+});
+```
+
+### Functions
+
+The standard library in Eve has been redone in the DSL. To use library functions, you must now bring in "lib" explicitly when defining your block. From lib you can access the various standard library functions supported by the runtime so far. 
+
+```javascript
+program.block("display the student's full names", ({find, record, lib}) => {
+  find("angle" degrees)
+  let result = math.sin(degrees)
+  return [
+    record("html/element", {tagname: "div", text: result})
+  ];
+});
+```
+
+for now you can find a list of functions in [src/runtime/stdlib.ts](https://github.com/witheve/Eve/blob/refactor-editor/src/runtime/stdlib.ts)
+
+The interface for wrapping functions for use within Eve is also revamped for the new runtime. When writing a function wrapper, you must ensure that the function meets the following qualifications
+
+1. The function must be referentially transparent, meaning given the same input, the function should return the same output.
+
+2. The function must be monotonic, meaning that additional inputs should never decrease the number of outputs.
+
+If the function meets these two qualifications, it can be wrapped using `makeFunction()` e.g.:
+
+```javascript
+makeFunction({
+  name: "math/sin",             
+  args: {a: "number"},
+  returns: {result: "number"},
+  apply: (a:number) => {
+    return [Math.sin(a/180 * Math.PI)];
+  }
+});
+```
+
+## Importing and Exporting Records 
+
+### Getting Data Into Eve - `inputEavs()`
+
+You can import raw EAVs into Eve with the `inputEavs()` function. Due to limitations of the host language, in order for it to work correctly, `inputEavs()` can only be used after you define all blocks.
```

Comment:
Due to another fixable limitation you need to call inputEavs at least once to kick the program off.